substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/SubstrateOptions.java

```diff
@@ -77,9 +79,38 @@ public static boolean parseOnce() {
     @Option(help = "Name of the main entry point method. Optional if --shared is used.")//
     public static final HostedOptionKey<String> Method = new HostedOptionKey<>("main");
 
+    @Option(help = "Directory of the image file to be generated", type = OptionType.User)//
+    public static final HostedOptionKey<String> Path = new HostedOptionKey<>(null);
+
     @Option(help = "Name of the output file to be generated", type = OptionType.User)//
     public static final HostedOptionKey<String> Name = new HostedOptionKey<>("");
 
+    public static String getImageDestinationDirectory() {
+        return getImageDestinationDirectory(HostedOptionValues.singleton());
+    }
+
+    public static String getImageDestinationDirectory(OptionValues values) {
+        if (Paths.get(Name.getValue(values)).getNameCount() > 1) {
+            return Paths.get(Path.getValue(values)).resolve(Name.getValue(values)).getParent().toString();
+        }
+        return Path.getValue(values);
+    }
+
+    public static String getImageName(OptionValues options) {
+        return Paths.get(Name.getValue(options)).getFileName().toString();
+    }
+
+    public static void verifyImagePath(OptionValues values) {
+        java.nio.file.Path path = Paths.get(Name.getValue(values));
+        UserError.guarantee(!path.isAbsolute(), "Image name may not be specified as an absolute path: %s%n", path);
```

Comment:
Why is that?

Reply:
Hmm, I guess due to https://github.com/oracle/graal/pull/3596#discussion_r671134286

I am still not sure why you can't handle absolute paths.